### PR TITLE
fix validator

### PR DESCRIPTION
**Pull request by ndoriomedov**

### CODE REVIEW

File: src/OfficeService.Models.Db/DbOffice.cs [hunk 1]
```diff
@@ -47,6 +51,14 @@ public void Configure(EntityTypeBuilder<DbOffice> builder)
         .Property(o => o.Address)
         .IsRequired();
 
+      builder
+        .Property(o => o.Latitude)
```
Comment (asyagraf) [inline src/OfficeService.Models.Db/DbOffice.cs]: is it necessary to use IsRequired() here? Latitude and Longitude are double, double can't be null

Comment (ndoriomedov) [reply src/OfficeService.Models.Db/DbOffice.cs:55]: yes, you right

**Review by VladSpitsin — APPROVED** [review]

File: src/OfficeService.Validation/Office/CreateOfficeRequestValidator.cs [hunk 2]
```diff
@@ -8,11 +8,20 @@ public class CreateOfficeRequestValidator : AbstractValidator<CreateOfficeReques
...
       RuleFor(request => request.City)
-        .Must(c => !string.IsNullOrEmpty(c?.Trim())).WithMessage("City must not be empty.");
+        .NotEmpty().WithMessage("City must not be empty.");
 
       RuleFor(request => request.Address)
-        .Must(a => !string.IsNullOrEmpty(a?.Trim())).WithMessage("Address must not be empty.");
+        .NotEmpty().WithMessage("Address must not be empty.");
+
+      RuleFor(request => request.Latitude)
+        .NotEmpty().WithMessage("Latitude must not be empty.");
```
Comment (asyagraf) [inline src/OfficeService.Validation/Office/CreateOfficeRequestValidator.cs]: Latitude and Longitude can't be empty or null as double, so NotEmpty() here means "!= default value for double". If this is what you meant then ok.

**Review by asyagraf — APPROVED** [review]

**Review by borovHokage — APPROVED** [review]

File: src/OfficeService.Business/Commands/Office/CreateOfficeCommand.cs [hunk 3]
```diff
@@ -52,14 +54,16 @@ public async Task<OperationResultResponse<Guid>> ExecuteAsync(CreateOfficeReques
         };
       }
 
-      if (!_validator.ValidateCustom(request, out List<string> errors))
+      ValidationResult validationResult = await _validator.ValidateAsync(request);
+
+      if (!validationResult.IsValid)
       {
         _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.BadRequest;
```
Comment (MariyaSin) [inline src/OfficeService.Business/Commands/Office/CreateOfficeCommand.cs]: why do you not use response creator?

File: src/OfficeService.Business/Commands/Office/EditOfficeCommand.cs [hunk 4]
```diff
@@ -56,14 +62,16 @@ public async Task<OperationResultResponse<bool>> ExecuteAsync(Guid officeId, Jso
         };
       }
 
-      if (!_validator.ValidateCustom(request, out List<string> errors))
+      ValidationResult validationResult = await _validator.ValidateAsync((officeId, request));
+
+      if (!validationResult.IsValid)
       {
         _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.BadRequest;
```
Comment (MariyaSin) [inline src/OfficeService.Business/Commands/Office/EditOfficeCommand.cs]: response creator

File: src/OfficeService.Data/Interfaces/IOfficeRepository.cs [hunk 5]
```diff
@@ -22,5 +22,9 @@ public interface IOfficeRepository
     Task<bool> EditAsync(Guid officeId, JsonPatchDocument<DbOffice> request);
 
     Task<bool> DoesExistAsync(Guid officeId);
+
+    Task<bool> IsNameUniqueAsync(string name);
```
Comment (MariyaSin) [inline src/OfficeService.Data/Interfaces/IOfficeRepository.cs]: may be doed name exist like other services?

File: src/OfficeService.Models.Db/DbOffice.cs [hunk 6]
```diff
@@ -39,6 +39,10 @@ public void Configure(EntityTypeBuilder<DbOffice> builder)
       builder
         .HasKey(o => o.Id);
 
+      builder
```
Comment (MariyaSin) [inline src/OfficeService.Models.Db/DbOffice.cs]: are you shure?

Comment (ndoriomedov) [reply src/OfficeService.Models.Db/DbOffice.cs]: No)

File: src/OfficeService.Business/Commands/Office/CreateOfficeCommand.cs [hunk 7]
```diff
@@ -24,43 +27,40 @@ public class CreateOfficeCommand : ICreateOfficeCommand
...
-        _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.Forbidden;
-
-        return new OperationResultResponse<Guid>
-        {
-          Status = OperationResultStatusType.Failed,
-          Errors = new() { "Not enough rights." }
-        };
+        _responseCreator.CreateFailureResponse<Guid>(
+          HttpStatusCode.Forbidden,
+          new() { "Not enough rights." });
```
Comment (MariyaSin) [inline src/OfficeService.Business/Commands/Office/CreateOfficeCommand.cs]: response creator contains the message for badRequest, Forbidden, no need write it

File: src/OfficeService.Business/Commands/Office/EditOfficeCommand.cs [hunk 8]
```diff
@@ -15,56 +18,57 @@
...
-        _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.Forbidden;
-
-        return new OperationResultResponse<bool>
-        {
-          Status = OperationResultStatusType.Failed,
-          Errors = new() { "Not enough rights." }
-        };
+        _responseCreator.CreateFailureResponse<bool>(
+          HttpStatusCode.Forbidden,
+          new() { "Not enough rights." });
```
Comment (MariyaSin) [inline src/OfficeService.Business/Commands/Office/EditOfficeCommand.cs]: same

File: src/OfficeService.Data/OfficeRepository.cs [hunk 9]
```diff
@@ -93,5 +93,15 @@ public async Task<List<DbOffice>> GetAsync(List<Guid> officesIds)
         .Include(o => o.Users).Where(u => u.IsActive)
         .ToListAsync();
     }
+
+    public async Task<bool> DoesNameExistAsync(string name)
+    {
+      return !await _provider.Offices.AnyAsync(x => x.IsActive && string.Equals(x.Name.ToLower(), name.ToLower()));
```
Comment (MariyaSin) [inline src/OfficeService.Data/OfficeRepository.cs]: check active and notActive offices

**Review by MariyaSin — APPROVED** [review]

**Pull request merged**